sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,577 @@
+        """
+
+        raise NotImplementedError
+
+    def activate_firmware(self, port, fwfile):
+        """ This routine should activate the downloaded firmware on all the
+        components of the Y cable of the port specified.
+        This API is meant to be used in conjunction with download_firmware API, and
+        should be called once download_firmware API is succesful.
+        This means that the firmware which has been downloaded should be
+        activated (start being utilized by the cable) once this API is
+        successfully executed.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+        Returns:
+            an Integer:
+                 a predefined code stating whether the firmware activate was successful
+        """
+
+        raise NotImplementedError
+
+    def rollback_firmware(self, port):
+        """ This routine should rollback the firmware to the previous version
+        which was being used by the cable. This API is intended to be called when the
+        user either witnesses an activate_firmware API failure or sees issues with
+        newer firmware in regards to stable cable functioning.
+
+        Args:
+            physical_port:
+                 an Integer, the actual physical port connected to a Y cable
+        Returns:
+            an Integer:
+                 a predefined code stating whether the firmware rollback was successful
+                 or an error code as to what was the cause of firmware rollback failure
+        """
+
+        raise NotImplementedError
+
+    def set_switching_mode(self, port, mode):
+        """
+        This API specifically enables the auto switching or manual switching feature on the muxcable,
+        depending upon the mode entered by the user.
+        Autoswitch feature if enabled actually does an automatic toggle of the mux in case the active
+        side link goes down and basically points the mux to the other side.
+
+        Args:
+             physical_port:
+                 an Integer, the actual physical port connected to Y end of a Y cable which can toggle the MUX
+             mode:
+                 an Integer, specifies which type of switching mode we set the muxcable to
+                 either SWITCHING_MODE_AUTO or SWITCHING_MODE_MANUAL
+
+        Returns:
+            a Boolean, true if the switch succeeded and false if it did not succeed.
+        """
+
+        raise NotImplementedError
+
+    def get_switching_mode(self, port):
+        """
+        This API specifically returns which type of switching mode the cable is set to auto/manual
+
+        Args:
+             physical_port:
+                 an Integer, the actual physical port connected to Y end of a Y cable which can toggle the MUX
+
+        Returns:
+            an Integer, SWITCHING_MODE_AUTO if auto switch is enabled.
+                        SWITCHING_MODE_MANUAL if manual switch is enabled.
```

Comment:
Maybe rather than explicitly stating "An Integer", we simply state "One of the following predefined constants:". This would also allow us to change the underlying type transparently if desired.

Reply:
fixed